documentation/Working_on_Bot/README.md

```diff
@@ -0,0 +1,257 @@
+
+Commands to Run
+```shell
+export CONTAINER_HOME_DIR=/usr/src/app;
+export COMPOSE_PROJECT_NAME="project_name"
+ ./CI/user_scripts/create-dev-docker-image.sh
+```
+
+#### Step 2. Launching the Bot
+```shell
+export ENVIRONMENT=LOCALHOST;
+//ensure that DB_ENBLED is set to 1 via whatever method you want
+export COMPOSE_PROJECT_NAME="project_name"
+
+# if you did not need to re-create the base image you can use
+export ORIGIN_IMAGE="sfucsssorg/wall_e"
+# otherwise, you will need to use the below commmand
+export ORIGIN_IMAGE="${COMPOSE_PROJECT_NAME}_wall_e_base"
+
+./CI/user_scripts/deploy-to-test-server.sh;
+```
+
+#### Re-launching the bot after making changes
+```shell
+
+## Stage 1: stop current bot
+# enter the command `.exit` on your discord guild
+
+## Stage 2: update base image
+# if you made any new changes to the either of the files specified below,
+ - wall_e/src/requirements.txt file
+ - CI/server_scripts/Dockerfile.base file
+# run the following commands:
+export CONTAINER_HOME_DIR=/usr/src/app;
+export COMPOSE_PROJECT_NAME="project_name"
+ ./CI/user_scripts/create-dev-docker-image.sh
+
+
+## Stage 3
+./CI/user_scripts/deploy-to-test-server.sh;
+```
+
+#### view logs in active time
+```shell
+ docker logs -f "${COMPOSE_PROJECT_NAME}_wall_e"
+```
+
+### Without the Database
+
+#### Step 1. Re-creating the docker base image
+You will need to recreate the base docker image if you
+ * made changes to the wall_e/src/requirements.txt file or
+ * made changes to the CI/server_scripts/Dockerfile.base file
+
+Commands To Run
+```shell
+export CONTAINER_HOME_DIR=/usr/src/app;
+export COMPOSE_PROJECT_NAME="project_name"
+./CI/user_scripts/create-dev-docker-image.sh
+```
+
+#### Step 2. Launching the Bot
+
+```shell
+export ENVIRONMENT=LOCALHOST;
+//ensure that DB_ENBLED is set to 0 via whatever method you want
+export COMPOSE_PROJECT_NAME="project_name"
+
+# if you did not need to re-create the base image you can use
+export ORIGIN_IMAGE="sfucsssorg/wall_e"
+# otherwise, you will need to use the below commmand
+export ORIGIN_IMAGE="${COMPOSE_PROJECT_NAME}_wall_e_base"
+
+./CI/user_scripts/deploy-to-test-server-nodb.sh;
+```
+
+
+#### Re-launching the bot after making changes
+```shell
+
+## Stage 1: stop current bot
+# enter the command `.exit` on your discord guild
+
+## Stage 2: update base image
+# if you made any new changes to the either of the files specified below,
+ - wall_e/src/requirements.txt file
+ - CI/server_scripts/Dockerfile.base file
+# run the following commands:
+export CONTAINER_HOME_DIR=/usr/src/app;
+export COMPOSE_PROJECT_NAME="project_name"
+ ./CI/user_scripts/create-dev-docker-image.sh
+
+
+ ## Stage 3
+./CI/user_scripts/deploy-to-test-server-nodb.sh;
+
+```
+
+#### view logs in active time
+```shell
+ docker logs -f "${COMPOSE_PROJECT_NAME}_wall_e"
+```
+
+## Testing the bot
+
+### Step 1. Run through the [linter](https://en.wikipedia.org/wiki/Lint_%28software%29)
```

Comment:
I think you can assume the reader knows what a linter is. You assume that they know what unit tests, git, docker, etc. are.

Reply:
I personally did not know what a linter actually was until this year myself so I don't think it hurts to have the link.

Reply:
But you've assumed knowledge of a lot of other tools when writing this README. It's inconsistent.

Reply:
For example, you assume knowledge of what an ini file is and Env variables. Why not assume knowledge of linters?

Reply:
fixed with 18e3c2283237e0b368c1429fb2b4635eaa5b5309